src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -73,18 +98,50 @@ public CommandResult execute(Model model) throws CommandException {
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
-
         Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
-        }
+        if (personToEdit instanceof Student) {
```

Comment:
just a minor point, but maybe you can abstract each if/else block into separate methods!